presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -823,6 +851,29 @@ private void assertSTGeometryN(String wkt, int index, String expected)
         assertFunction("ST_ASText(ST_GeometryN(ST_GeometryFromText('" + wkt + "')," + index + "))", VARCHAR, expected);
     }
 
+    @Test
+    public void testSTInteriorRingN()
+    {
+        assertInvalidFunction("ST_AsText(ST_InteriorRingN(ST_GeometryFromText('POINT EMPTY'), 0))", "ST_InteriorRingN only applies to POLYGON. Input type is: POINT");
+        assertInvalidFunction("ST_AsText(ST_InteriorRingN(ST_GeometryFromText('LINESTRING (1 2, 2 3, 3 4)'), 1))", "ST_InteriorRingN only applies to POLYGON. Input type is: LINE_STRING");
+        assertInvalidFunction("ST_AsText(ST_InteriorRingN(ST_GeometryFromText('MULTIPOINT (1 1, 2 3, 5 8)'), -1))", "ST_InteriorRingN only applies to POLYGON. Input type is: MULTI_POINT");
+        assertInvalidFunction("ST_AsText(ST_InteriorRingN(ST_GeometryFromText('MULTILINESTRING ((2 4, 4 2), (3 5, 5 3))'), 0))", "ST_InteriorRingN only applies to POLYGON. Input type is: MULTI_LINE_STRING");
+        assertInvalidFunction("ST_AsText(ST_InteriorRingN(ST_GeometryFromText('MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((2 4, 2 6, 6 6, 6 4)))'), 2))", "ST_InteriorRingN only applies to POLYGON. Input type is: MULTI_POLYGON");
+        assertInvalidFunction("ST_AsText(ST_InteriorRingN(ST_GeometryFromText('GEOMETRYCOLLECTION (POINT (2 2), POINT (10 20))'), 1))", "ST_InteriorRingN only applies to POLYGON. Input type is: GEOMETRY_COLLECTION");
+
+        assertSTInteriorRingN("POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", 1, null);
+        assertSTInteriorRingN("POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", 2, null);
+        assertSTInteriorRingN("POLYGON ((0 0, 1 0, 1 1, 0 1, 0 0))", -1, null);
```

Comment:
Is this a standard? Why would you need to silently accept such indexes?

Reply:
@arhimondr Yes, this is the behavior specified in SQL/MM Part 3 standard

```
If aposition is less than one or greater than the cardinality of the ST_PrivateInteriorRings attribute, then:
i) A completion condition is raised: SQL/MM Spatial warning – invalid position. 
ii) Return the null value.
```

Reply:
I'm following the behavior shown in ST_GeometryN, which returns null if the the index is out of bounds. I believe this is standard.